### PR TITLE
Port over favors greedy and modest to autofixture nunit3

### PR DESCRIPTION
**Pull request by hackle**
Question - should [FavorLists] and [FavorEnumberables] be included in the Scenarios?

### CODE REVIEW

File: Src/AutoFixture.NUnit3.UnitTest/Scenario.cs [hunk 1]
```diff
@@ -65,6 +65,33 @@ public void FreezeSecondParameterOnlyFreezesSubsequentParameters(Guid g1, [Froze
...
+
+        [Test, AutoData]
+        public void GreedyCreatesParameterWithGreedyConstructor([Greedy]MultiUnorderedConstructorType p)
+        {
+            Assert.False(string.IsNullOrEmpty(p.Text));
+            Assert.AreNotEqual(0, p.Number);
+        }
+
+        [Test, AutoData]
+        public void BothFrozenAndGreedyAttributesCanBeAppliedToSameParameter([Frozen][Greedy]MultiUnorderedConstructorType p1, MultiUnorderedConstructorType p2)
```
Comment (ploeh) [inline Src/AutoFixture.NUnit3.UnitTest/Scenario.cs]: In .NET, the order of attributes are not guaranteed. What happens if `[Frozen]` is evaluated and executed _before_ `[Greedy]`? Could that make the test fail?

**Comment by hackle** [comment]
Added another unit test with [Frozen] and [Greedy] in reverse order - however that relies on .NET being consistent for one execution, e.g. it's not doing a Random() or Shuffle() for the attributes, which even true I doubt should be covered.

**Comment by ploeh** [comment]
Why is `TestResult.xml` checked in? Where does it come from?

I'd have hoped that #563 would have ignored such a file, but I can see that it doesn't match that particular pattern.

**Comment by hackle** [comment]
My bad, I think by mistake I must have checked in the TestResult.xml which comes out of a manual run of the NUnit3 Console. The build script specified a result file name so there is nothing wrong with it.

Thanks @ploeh

**Comment by ploeh** [comment]
Thank you for your contribution! It's now live as [AutoFixture.NUnit3 3.43.0](https://www.nuget.org/packages/AutoFixture.NUnit3/3.43.0).

**Pull request merged**